test/Microsoft.AspNetCore.Owin.Tests/OwinExtensionTests.cs

```diff
@@ -63,25 +67,26 @@ public void OwinDefaultNoServices()
             {
                 builderExecuted = true;
                 serviceProvider = applicationBuilder.ApplicationServices;
-                applicationBuilder.Run(async context =>
+                applicationBuilder.Run(context =>
                 {
                     applicationExecuted = true;
                     fakeService = context.RequestServices.GetService<FakeService>();
+                    return Task.FromResult(0);
                 });
-            });
+            },
+            expectedServiceProvider);
```

Comment:
I thought it was to not add the `FakeService` to the service provider. Without a service provider, the delegate `throw`s an NRE.